src/module.c

```diff
@@ -4074,7 +4074,7 @@ int RM_StringTruncate(RedisModuleKey *key, size_t newlen) {
             sdssubstr(key->value->ptr,0,newlen);
             /* If the string is too wasteful, reallocate it. */
             if (sdslen(key->value->ptr) < sdsavail(key->value->ptr))
-                key->value->ptr = sdsRemoveFreeSpace(key->value->ptr);
+                key->value->ptr = sdsRemoveFreeSpace(key->value->ptr, 0);
```

Comment:
maybe in this case we rather pass 1?
in the past calling sdsRemoveFreeSpace would have a chance to use TYPE_5, so passing 0 would keep the old behavior. but i'm not sure we know how modules use it, maybe they do intend to re-grow later?

@MeirShpilraien do you have any feedback on how you think modules use either RM_TrimStringAllocation or RM_StringTruncate?

Reply:
another related thought, when we introduced RM_TrimStringAllocation (calls trimStringObjectIfNeeded), we thought of trimming argv strings (as an alternative to the automatic trimming done in RedisModuleCommandDispatcher, we even documented it.
but maybe modules also call it on non-argv strings? in which case the new condition on BIG_ARG would cause damage?

Reply:
`TestStringAppendAM()` test is to use `RM_TrimStringAllocation()` for non-argv strings, maybe we could add a minimum reserved parameter to trimStringObjectIfNeeded(), if it's 0 then there is no limit.

Reply:
> @MeirShpilraien do you have any feedback on how you think modules use either RM_TrimStringAllocation or RM_StringTruncate?

@oranagra I do not think any of our modules use it.

Reply:
it's hard to make a decision here, but i have a feeling the right thing is to keep the code as is.

i.e. assume that RM_TrimStringAllocation is used on argv strings, when deciding to store them in the db.
keep it's capability to use SDS_TYPE_5 since it's likely to be used on strings that are not gonna mutate (same goes for RM_StringTruncate, let's let it keep it's possibility for using TYPE_5 for now),

Also let's keep the condition that avoid extra work on small strings for RM_TrimStringAllocation, since if they are coming from argv, small ones will have no excess.

it's already documented in the module api what is it's purpose, so the only thing left is to document in the top comment of the PR the change it causes this API (not trimming small strings, in case they're not coming from argv)